### PR TITLE
DSG-2327: Multi file progress and uploading status

### PR DESCRIPTION
**Pull request by kevincheng0**
# Basics
Added uploading progress bar, docai findings, and cancellation for multiple files

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature
- [ ] Styling and layout changes
- [ ] Code style changes (formatting, local variables)
- [ ] Refactoring (no functional changes)
- [ ] Build or CI related changes
- [ ] Tests
- [ ] Other... Please describe:

## What changes does this PR introduce?
- Added new component file-upload-list to house the new list for multiple files (to lower cyclomatic complexity)
- Refactored progress bar for multiple files
- Refactored file processing
- Refactored cancellation

## Screenshot of the affected changes
[Screencast from 11-23-2023 03:05:31 PM.webm](https://github.com/Nuvalence/dsgov-web/assets/65138214/d84e4e61-7a9d-49d8-8673-c0a4f9a49e04)

## Checklist

- [ ] I have updated the documentation accordingly
- [ ] I have confirmed all acceptance criteria defined by the story have been met
- [ ] I have confirmed that Sonar Qube findings for new code are reviewed and fixed, or will be whitelisted

### CODE REVIEW

File: libs/shared/ui/nuverial/src/lib/components/file-upload/file-upload.component.ts [hunk 1]
```diff
@@ -417,7 +473,7 @@ export class NuverialFileUploadComponent extends FormInputBaseDirective implemen
         const fileSizeError = this.validateFileSizeError();
         const errors = fileSizeError || this.validateRequired();
         if (fileSizeError) {
-          this.stopUpload();
+          this.stopUpload('');
```
Comment (derekdiep42) [inline libs/shared/ui/nuverial/src/lib/components/file-upload/file-upload.component.ts:476]: what is this here for? stopUpload doesn't do anything if you send an empty name right?

**Review by derekdiep42 — COMMENTED** [review]
There's a lot of logic here, so I'm going to leave it to @davidyoung77 when he comes back for a more in-depth review. But nice work!

File: libs/shared/ui/nuverial/src/lib/components/file-upload-list/file-upload-list.component.html [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+        class="file-uploaded__label__icon--{{file.status}}"
+        [iconName]="file.status === statusOptions.success ? 'check_circle' : file.status === statusOptions.failure ? 'error' : 'refresh'"
+        [outlined]="true"
+      ></nuverial-icon>
+      <span class="file-uploaded__label__name">{{file.name}}</span>
+    </div>
+    <div class="file-uploaded__actions">
+      <div class="file-uploaded__actions__tooltip" *ngIf="file.status === statusOptions.failure && file.processingStatus.processors?.length">
+        <nuverial-file-processor-tooltip [processors]="file.processingStatus.processors">
+          <span class="file-uploaded__actions__tooltip__text">Document Issues Detected</span>
```
Comment (davidyoung77) [inline libs/shared/ui/nuverial/src/lib/components/file-upload-list/file-upload-list.component.html:38]: The text here `Document issues detected` is inconsistent with the single file implementation. can you check with @czara5 to see if we can make this consistent

**Review by davidyoung77 — APPROVED** [review]
This looks good, I have a product comment about implementation

File: libs/shared/ui/nuverial/src/lib/components/file-upload-list/file-upload-list.component.html (hunk 2 above)
Comment (kevincheng0) [reply libs/shared/ui/nuverial/src/lib/components/file-upload-list/file-upload-list.component.html:38]: Alright, I'll ask Czara about the text here

Comment (kevincheng0) [reply libs/shared/ui/nuverial/src/lib/components/file-upload-list/file-upload-list.component.html:38]: Yes, we're changing both to `Document Issues Detected`

**Review by derekdiep42 — APPROVED** [review]

**Pull request merged**